code/game/machinery/slotmachine.dm

```diff
@@ -54,23 +61,41 @@ var/datum/announcement/minor/slotmachine_announcement = new(do_log = 0)
 					playsound(src.loc, 'sound/effects/engine_alert2.ogg', 50, 0)
 					visible_message("<b>[src]</b> says, 'JACKPOT! [usr.name] has won two hundred and fifty thousand credits!'")
 					slotmachine_announcement.Announce("Congratulations to [usr.name] on winning the jackpot of two hundred and fifty thousand credits!", "Jackpot Winner")
-					usr.mind.initial_account.money += 250000
+					result = "JACKPOT! You win two hundred and fifty thousand credits!"
+					resultlvl = "highlight"
+					win_money(250000, 0)
 				else if (roll > 1 && roll <= 10)
-					playsound(src.loc, 'sound/machines/ping.ogg', 50, 0)
 					visible_message("<b>[src]</b> says, 'Big Winner! [usr.name] has won five thousand credits!'")
-					usr.mind.initial_account.money += 5000
+					result = "Big Winner! You win five thousand credits!"
+					resultlvl = "good"
+					win_money(5000)
 				else if (roll > 10 && roll <= 100)
-					playsound(src.loc, 'sound/machines/ping.ogg', 50, 0)
-					visible_message("<b>[src]</b> says, 'Winner! [usr.name] has won win five hundred credits!'")
-					usr.mind.initial_account.money += 500
+					visible_message("<b>[src]</b> says, 'Winner! [usr.name] has won five hundred credits!'")
+					result = "You win five hundred credits!"
+					resultlvl = "good"
+					win_money(500)
 				else if (roll > 100 && roll <= 1000)
-					playsound(src.loc, 'sound/machines/ping.ogg', 50, 0)
-					usr << "<span class='notice'>You win 5 credits!</span>"
-					usr.mind.initial_account.money += 5
+					result = "You win 5 credits!"
+					resultlvl = "good"
+					win_money(5)
 				else
-					usr << "<span class='warning'>No luck!</span>"
+					result = "<span class='warning'>No luck!</span>"
+					resultlvl = "average"
 				working = 0
 				icon_state = "slots-off"
-	add_fingerprint(usr)
-	updateUsrDialog()
-	return
+
+/obj/machinery/slot_machine/proc/win_money(amt, playsnd=1)
```

Comment:
I could replace that var, since it's just there so that the jackpot could do its own sound anyway, would make the jackpot cleaner.